samples/subsys/mgmt/osdp/control_panel/sample.yaml

```diff
@@ -4,7 +4,7 @@ sample:
 tests:
   sample.mgmt.osdp.control_panel:
     tags: osdp
-    depends_on: gpio
+    depends_on: gpio arduino_serial
     filter: dt_enabled_alias_with_parent_compat("led0", "gpio-leds") and
             dt_chosen_enabled("zephyr,osdp-uart") and CONFIG_SERIAL
```

Comment:
filter is applied before dtc compilation, hence before the overlay is applied.
`dt_chosen_enabled("zephyr,osdp-uart")` will always return false.

Reply:
So how can I filter platforms that aren't supported? removing `dt_chosen_enabled("zephyr,osdp-uart")` widens the test space and so the test fails on unsupported platforms.

Reply:
Well, in theory "filter" should list allow to filter out al platforms that don"'t provide required dependencies, so in theory it shouldn't.
But I agree that on this area, practice often defeats the theory.  Out of curiosity, do you have an example of unsupported platform that is not filtered out ?

Reply:
Just removing `dt_chosen_enabled("zephyr,osdp-uart")` from the filter causes test failure because the device tree lookup for "zephyr,osdp-uart".

I don't understand the failures as the app overlays should be applied at the time (after this filter is processed).

Reply:
Can you provide an example of failing test command so I can have a try?

Reply:
Actually that is part of the problem; I cannot run the tests locally because something is broken in my python setup. Haven't spent the time fixing it.

Removing `dt_chosen_enabled("zephyr,osdp-uart")` caused CI builds to fail. I'll do that now.

Reply:
@erwango, can you help me with this CI failure please :)